platform/ios/src/MGLMapView.mm

```diff
@@ -1566,7 +1604,11 @@ - (void)handleDoubleTapGesture:(UITapGestureRecognizer *)doubleTap
             [self animateWithDelay:MGLAnimationDuration animations:^
              {
                  [weakSelf unrotateIfNeededForGesture];
+                 [weakSelf notifyScaleBarGestureDidEnd];
              }];
+        } else {
+            [self unrotateIfNeededForGesture];
+            [self notifyScaleBarGestureDidEnd];
```

Comment:
In the past, atomic gestures like double-tapping avoided calling `-notifyGestureDidBegin` and `-notifyGestureDidEndWithDrift:`. We should have these gestures call `-notifyGestureDidBegin` and `-notifyGestureDidEndWithDrift:` one right after the other, and merge `-notifyScaleBarGestureDidEnd` into `-notifyGestureDidEndWithDrift:`. That way there are fewer things to keep track of every time we finish a gesture.